### PR TITLE
Migrate from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by jasonwilliams**
As travis-ci is going away this migrates CI (on both pull request and pushes to master) to Github Actions.
It may take a bit of trial and error, but hopefully what I have here should work.

There's currently a failing test so I expect it to fail but it should help highlight issues. The failing test is being addressed [here](https://github.com/microsoft/typescript-styled-plugin/pull/128)
@mjbvz

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+on:
+    push:
+        branches:
+            - master
```
Comment (mjbvz) [inline .github/workflows/ci.yml:4]: We've renamed the primary branch from `master` to `main` in this repo

**Review by mjbvz — CHANGES_REQUESTED** [review]
Thanks for looking into this. Just one minor thing about branch names, otherwise looks good

**Comment by jasonwilliams** [comment]
@mjbvz updated the branch name

**Comment by mjbvz** [comment]
Thanks!

**Pull request merged**